### ISSUE TITLE
Getting missing Form label for mat-list-option for angular material 15 after MDC migration

### ISSUE THREAD

**Issue by shwetanikam12** [angular/material.angular.io#1221]
- After angular material upgrade to version 15 and MDC upgrade we are getting an accessibility error from the WAVE tool. 
- We are getting **_missing-form-label_** error for every checkbox for the **_mat-list-option_**. We have tried adding aria-label ,alt-text, title and even tried adding mat-label inside the **_mat-list-option_**, nothing has been worked till yet.
- Inside <mat-list-option> input field is getting created on the fly for which this error is coming.
- We have found one solution in which with the help of **_DOM_** manipulation we are able to resolve this issue, since DOM manipulation is not recommended is there some other solution for it.
- This issue was not present before version 15 upgrade, even on the official material website we are able to see this issue for **_mat-selection-list_**.

Angular Material Version 14:
<img width="945" alt="image" src="https://github.com/angular/material.angular.io/assets/61000457/7798051a-0054-46c9-b22a-5594b628aba5">

Angular Material Version 15:
<img width="956" alt="image" src="https://github.com/angular/material.angular.io/assets/61000457/7a6423f3-046a-47da-acb1-512ddcdd1970">

**Comment by DhivyaDelphina** [comment]
any update on the above issue?
